diffpy/snmf/subroutines.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+
+
+# import scipy.interpolate
+
+
+def objective_function(residual_matrix, stretching_factor_matrix, smoothness, smoothness_term, component_matrix,
+                       sparsity):
+    """Defines the objective function of the algorithm and returns its value
+
```

Comment:
leave one blank line then add more explanation by actually defining the function (i.e., lines 47, 48) here in the docstring.

Reply:
Ok. I will do this. Should I add a reference to the paper?

Reply:
Good impulse.  However, when we write the documentation (and the readme) we will reference the paper and discuss it a bit, but it doesn't do any harm to put the ref here too, though the paper is not published yet, so maybe not. or we will have lots of out of date references hanging around in the code.